op-node/rollup/finality/finalizer.go

```diff
@@ -245,11 +245,13 @@ func (fi *Finalizer) tryFinalize() {
 	// go through the latest inclusion data, and find the last L2 block that was derived from a finalized L1 block
 	for _, fd := range fi.finalityData {
 		if fd.L2Block.Number > finalizedL2.Number && fd.L1Block.Number <= fi.finalizedL1.Number {
-			shouldContinue, err := tryFinalizeOnConsecutiveBlockQuorom(fd, fi, &finalizedL2, &finalizedDerivedFrom)
+			lastFinalizedBlockNumber, err := tryFinalizeOnConsecutiveBlockQuorom(fd, fi, &finalizedL2, &finalizedDerivedFrom)
```

Comment:
let's add a commnet here that this function modifies the two pointer arguments: finalizedL2 and finalizedDerivedFrom

I am reading it and already got confused and realized after digging into the code.

Reply:
refactor it